src/javascript/crypto/e2e/vrf/sha3shake256.js

```diff
@@ -0,0 +1,353 @@
+/**
+ * @license
+ * Copyright 2015 Yahoo Inc. All rights reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview Extends js-sha3 to support SHAKE256, and outputs byte array
+ */
+
+// var e2e = {vrf:module.exports};
+goog.provide('e2e.vrf.sha3Shake256');
+
+goog.require('goog.asserts');
+
+/*
+ * Imported from js-sha3 v0.3.1
+ * https://github.com/emn178/js-sha3
+ *
+ * Copyright 2015, emn178@gmail.com
+ * 
+ * Licensed under the MIT license:
+ * http://www.opensource.org/licenses/MIT
```

Comment:
Sorry for the delay. There are a couple of ways to address this, and it's mostly driven by project convention. Some projects have a strict process where the headers are kept consistent and that all inclusions like this are disclosed in a NOTICES file, other projects are much less strict and have what you see above -- an inclusion in the code that is preceded with the relevant license and copyright text. Both are OK. At first this latter approach is simpler, eventually the code could get messy and you'll want to clean it up. 

What is most important though is that somewhere in the project you indicate that emn17 owns the copyright of some code that was licensed under the MIT license and is included in this project.